contracts/cw1-subkeys/src/contract.rs

```diff
@@ -239,11 +263,115 @@ mod tests {
+        let spenders = vec![spender1.clone(), spender2.clone()];
+
+        // Same allowances for all spenders, for simplicity
+        let denom1 = "token1";
+        let denom2 = "token2";
+        let amount1 = 1111;
+        let amount2 = 2222;
+        let allowances = vec![coin(amount1, denom1), coin(amount2, denom2)];
+
+        let expires_height = Expiration::AtHeight { height: 5432 };
+        let expires_never = Expiration::Never {};
+        let expires_time = Expiration::AtTime { time: 1234567890};
+        // Initially set first spender allowance with height expiration, the second with no expiration
+        let expirations = vec![expires_height.clone(), expires_never.clone()];
+
+        let allow1 = &allowances[0];
+        let allow2 = &allowances[1];
+
+        let env = mock_env(owner, &[]);
+        setup_test_case(
+            &mut deps,
+            &env,
+            &admins,
+            &spenders,
+            &allowances,
+            &expirations,
+        );
+
+        // Add to spender1 account (expires = None) => don't change Expiration
+        let msg = HandleMsg::IncreaseAllowance {
+            spender: spender1.clone(),
+            amount: allow1.clone(),
+            expires: None,
+        };
+        handle(&mut deps, env.clone(), msg).unwrap();
+
+        // Verify
+        let allowance = query_allowance(&deps, spender1.clone()).unwrap();
+        assert_eq!(
+            allowance,
+            Allowance {
+                balance: Balance(vec![coin(amount1 * 2, &allow1.denom), allow2.clone()]),
+                expires: expires_height.clone()
+            }
+        );
+
+        // Add to spender2 account (expires = Some)
+        let msg = HandleMsg::IncreaseAllowance {
+            spender: spender2.clone(),
+            amount: allow1.clone(),
+            expires: Some(expires_height.clone()),
+        };
+        handle(&mut deps, env.clone(), msg).unwrap();
+
+        // Verify
+        let allowance = query_allowance(&deps, spender2.clone()).unwrap();
+        assert_eq!(
+            allowance,
+            Allowance {
+                balance: Balance(vec![coin(amount1 * 2, &allow1.denom), allow2.clone()]),
+                expires: expires_height.clone()
+            }
+        );
+
+        // Add to spender3 (new account) (expires = None) => default Expiration::Never
+        let msg = HandleMsg::IncreaseAllowance {
+            spender: spender3.clone(),
+            amount: allow1.clone(),
+            expires: None,
+        };
+        handle(&mut deps, env.clone(), msg).unwrap();
+
+        // Verify
+        let allowance = query_allowance(&deps, spender3.clone()).unwrap();
+        assert_eq!(
+            allowance,
+            Allowance {
+                balance: Balance(vec![allow1.clone()]),
+                expires: expires_never.clone()
+            }
+        );
+
+        // Add to spender4 (new account) (expires = Some)
```

Comment:
Nicely covering all the use cases.

It actually seems the `zip` above did exactly what you wanted it to do. 
but `setup_test_case` needs better rustdoc, so this is expected behavior

Reply:
I think it's related to the  comment above. I'll clarify the code with better naming / comments.